docs/api/graphql/graphql_custom_ft.md

```diff
@@ -0,0 +1,163 @@
+|------|------| ------- | ------- |
+| Text Line     | string | default | `TextLineFieldDefinition`    |
+| Relation List | `Item` `ArticleItem` `ImageItem` | customized | `RelationListFieldDefinitio` |
+
+## Map a custom Field Type
+
+There are two ways to map a custom Field Type:
+
+- configuration
+- custom `FieldDefinitionMapper`
+
+If the mapping of your Field Type depends on the field definition, you need to write a custom `FieldDefinitionMapper`.
+Otherwise, you can map with configuration.
+
+### Map with configuration
+
+To map a custom Field Type with configuration use a compiler pass to modify a container parameter, `ibexa.graphql.schema.content.mapping.field_definition_type`.
+
+It is a hash that maps a Field Type identifier (`ezstring`) to the following entries:
+
+- `value_type` - the GraphQL type values of custom field. It can be a native type (string, int), or a custom type. If none is specified, string will be used.
+- `value_resolver` - how values of this field are resolved and passed to the defined value type. If not specified, it will receive the `Field` object for the field type: `field`.
+- `definition_type` - the GraphQL type field definitions. If not specified, it will use `FieldDefinition`.
+
+Compiler pass example that should be placed in `src/DependencyInjection/Compiler`:
+
+``` php
+[[= include_file('code_samples/api/graphql/src/DependencyInjection/Compiler/MyCustomTypeGraphQLCompilerPass.php') =]]
+```
+
+### Map with a custom `FieldDefinitionMapper`
+
+The `FieldDefinitionMapper` API uses service decorators.
+To register your own mapper, make it decorate the
+`Ibexa\GraphQL\Schema\Domain\Content\Mapper\FieldDefinition\DecoratingFieldDefinitionMapper` service:
+
+```yaml
+services:
+    App\GraphQL\Schema\MyCustomFieldDefinitionMapper:
+        decorates: Ibexa\GraphQL\Schema\Domain\Content\Mapper\FieldDefinition\DecoratingFieldDefinitionMapper
+        arguments:
+            $innerMapper: '@.inner'
+```
+
+The `$innerMapper` argument will pass the decorated mapper to the constructor.
+You can use the `DecoratingFieldDefinitionMapper` from the `graphql` package.
+It requires that you implement the `getFieldTypeIdentifier` method to tell which Field Type is covered by the mapper.
+
+Add `MyCustomFieldDefinitionMapper.php` mapper to `src/GraphQL/Schema`:
+
+``` php
+[[= include_file('code_samples/api/graphql/src/GraphQL/Schema/MyCustomFieldDefinitionMapper.php') =]]
+```
+
+The `FieldDefinitionMapper` interface defines three methods:
+
+- `mapToFieldValueType` - returns the GraphQL type value for the defined field
+- `mapToFieldValueInputType` - returns the GraphQL type value for the field input value
+- `mapToFieldValueResolver` - returns the resolver, as an expression language string, values are resolved with
+- `mapToFieldDefinitionType`- returns the GraphQL type field definitions of the mapped type
+
+Only implement methods that you need, the rest will be handled by other mappers (configuration or default).
```

Comment:
We need to add a note about the need to call this (with mapXXX replaced by the method it is in):

```
        if (!$this->canMap($fieldDefinition)) {
             return parent::mapToFieldValueType($fieldDefinition);
         }
```

It is required in every method that is implemented, so that other mappers are called for other field types.